lib/mixins/property-effects.js

```diff
@@ -2390,6 +2390,15 @@ export const PropertyEffects = dedupingMixin(superClass => {
      * @protected
      */
     static _addTemplatePropertyEffect(templateInfo, prop, effect) {
+      // `dynamicFns` is the flattened property list, so we can use that to
+      // detect non-declared properties.  Properties must be listed in
+      // `properties` to be included in `observedAttributes` since CE V1
+      // reads that at registration time, and we want to keep template parsing
+      // lazy
+      if (legacyOptimizations && !(prop in templateInfo.dynamicFns)) {
```

Comment:
Using `dynamicFns` here seems a bit abusive since there's no guarantee this is connected to `observedAttributes`. One alternative would be to override this in `element-mixin` and use `this._properties`.

Reply:
Moved to property-effects.